src/quo2/components/selectors/selectors.cljs

```diff
@@ -68,46 +69,51 @@
                      (colors/alpha colors/neutral-100 (if disabled? 0.3 1))
                      (colors/alpha colors/white (if disabled? 0.3 1)))}]])]])))
 
-
 (defn checkbox
-  [{:keys [default-checked? on-change disabled? blurred-background? container-style]}]
-  (let [checked? (reagent/atom (or default-checked? false))]
-    [rn/touchable-without-feedback
-     {:on-press (handle-press disabled? on-change checked?)}
-     [rn/view
-      {:style (merge
-               container-style
-               {:height 20
-                :width  20})}
-      [rn/view
-       {:style               {:flex             1
-                              :border-radius    6
-                              :border-width     (if @checked? 0 1)
-                              :background-color (cond
-                                                  @checked?
-                                                  (get-color @checked? disabled? blurred-background?)
-                                                  blurred-background?
-                                                  (colors/theme-colors
-                                                   colors/white-opa-5
-                                                   colors/white-opa-10)
-                                                  :else
-                                                  (colors/theme-colors
-                                                   colors/white
-                                                   colors/neutral-80-opa-40))
-                              :border-color     (if @checked?
-                                                  :none
-                                                  (get-color @checked? disabled? blurred-background?))}
-        :accessibility-label (str "checkbox-" (if @checked? "on" "off"))
-        :accessibility-role  :checkbox
-        :testID              "checkbox-component"}
-       (when @checked?
+  [{:keys [default-checked?]}]
+  (let [internal-checked? (reagent/atom (or default-checked? false))]
+    (fn [{:keys [on-change disabled? blurred-background? container-style checked?]}]
+      (when (and (not (nil? checked?)) (not= @internal-checked? checked?))
+        (reset! internal-checked? checked?))
+      (let [checked? (or checked? @internal-checked?)]
+        [rn/touchable-without-feedback
+         {:on-press (handle-press disabled? on-change internal-checked?)}
          [rn/view
-          {:style
-           {:height 20
-            :width  20}}
-          [icons/icon :i/check-small
-           {:size  20
-            :color colors/white}]])]]]))
+          {:style (merge
+                   container-style
+                   {:height 20
+                    :width  20})}
+          [rn/view
+           {:style               {:flex             1
```

Comment:
would be cool to move styles to style ns, i know not related but as we touch this part anyway